Sources/Paywalls/PaywallColor.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  PaywallColor.swift
+//  
+//
+//  Created by Nacho Soto on 7/14/23.
+//
+
+import Foundation
+
+#if canImport(SwiftUI)
+import SwiftUI
+#endif
+
+#if canImport(UIKit)
+import UIKit
+#endif
+
+// swiftlint:disable redundant_string_enum_value
+
+/// Represents a color to be used by `RevenueCatUI`
+public struct PaywallColor {
+
+    /// The possible color schemes, corresponding to the light and dark appearances.
+    @frozen
+    public enum ColorScheme: String {
+
+        /// The color scheme that corresponds to a light appearance.
+        case light = "light"
+        /// The color scheme that corresponds to a dark appearance.
+        case dark = "dark"
+
+    }
+
+    /// The original Hex representation for this color.
+    public var stringRepresentation: String
+
+    /// The underlying SwiftUI `Color`.
+    @available(iOS 13.0, tvOS 13.0, macOS 10.15, watchOS 6.2, *)
+    public var underlyingColor: Color {
+        // swiftlint:disable:next force_cast
+        return self._underlyingColor as! Color
+    }
+
+    // Only available from iOS 3
+    fileprivate var _underlyingColor: (any Sendable)?
```

Comment:
We can’t make the type itself have `@available` because it’s deep inside `Offering`        `